contracts/external/cw-payroll/src/state.rs

```diff
@@ -0,0 +1,122 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, DepsMut, StdResult, Storage, Timestamp, Uint128};
+use cw_storage_plus::{Item, Map};
+use serde::{Deserialize, Serialize};
+
+use crate::{balance::WrappedBalance, msg::StreamId};
+
+#[cw_serde]
+pub struct Config {
+    pub admin: Addr,
+}
+
+pub const CONFIG: Item<Config> = Item::new("config");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct Stream {
+    pub admin: Addr,
+    pub recipient: Addr,
+    /// Balance in Native and Cw20 tokens
+    pub balance: WrappedBalance,
+    pub claimed_balance: WrappedBalance,
+    pub start_time: u64,
+    pub end_time: u64,
+    pub paused_time: Option<u64>,
+    pub paused_duration: Option<u64>,
+    pub paused: bool,
+
+    // METADATA
+    /// Title of the payroll item, for example for a bug bounty "Fix issue in contract.rs"
+    pub title: Option<String>,
+    /// Description of the payroll item, a more in depth description of how to meet the payroll conditions
+    pub description: Option<String>,
+    /// Link to stream attached for sync
+    pub link_id: Option<StreamId>,
+    /// Making a stream detachable will only affect linked streams.
+    /// A linked stream that detaches in the future will pause both streams.
+    /// Each stream must then resume on their own, or be fully removed to re-link.
+    pub is_detachable: bool,
+}
+
+impl Stream {
+    pub(crate) fn can_distribute_more(&self) -> bool {
+        if self.balance.amount() == 0 {
+            return false;
+        }
+        self.claimed_balance.amount() < self.balance.amount()
+    }
+    pub(crate) fn calc_distribution_rate_core(
+        start_time: u64,
+        end_time: u64,
+        block_time: Timestamp,
+        paused_duration: Option<u64>,
+        balance: &WrappedBalance,
+        claimed: &WrappedBalance,
+    ) -> (Uint128, Uint128) {
+        let block_time = std::cmp::min(block_time.seconds(), end_time);
+        let duration: u64 = end_time.checked_sub(start_time).unwrap_or_default();
```

Comment:
why do we `unwrap_or_default()` in this function instead of returning an error? a panic would be fine here if it's because we don't expect this to happen for valid streams in state

Reply:
Yeah we should return an error here.